packages/opentelemetry-plugin-grpc-js/src/client/utils.ts

```diff
@@ -0,0 +1,243 @@
+          [RpcAttribute.GRPC_ERROR_NAME]: err.name,
+          [RpcAttribute.GRPC_ERROR_MESSAGE]: err.message,
+        });
+      } else {
+        span.setStatus({ code: CanonicalCode.OK });
+        span.setAttribute(
+          RpcAttribute.GRPC_STATUS_CODE,
+          CanonicalCode.OK.toString()
+        );
+      }
+
+      span.end();
+      callback(err, res);
+    };
+    return plugin.tracer.bind(wrappedFn);
+  }
+
+  return (span: Span) => {
+    const metadata = getMetadata.call(plugin, original, args);
+    // if unary or clientStream
+    if (!original.responseStream) {
+      const callbackFuncIndex = findIndex(args, arg => {
+        return typeof arg === 'function';
+      });
+      if (callbackFuncIndex !== -1) {
+        args[callbackFuncIndex] = patchedCallback(
+          span,
+          args[callbackFuncIndex] as SendUnaryDataCallback<ResponseType>
+        );
+      }
+    }
+
+    span.setAttributes({
+      [RpcAttribute.GRPC_METHOD]: original.path,
+      [RpcAttribute.GRPC_KIND]: SpanKind.CLIENT,
+    });
+
+    setSpanContext(metadata);
+    const call = original.apply(self, args);
+
+    // if server stream or bidi
+    if (original.responseStream) {
+      // Both error and status events can be emitted
+      // the first one emitted set spanEnded to true
+      let spanEnded = false;
+      const endSpan = () => {
+        if (!spanEnded) {
+          span.end();
+          spanEnded = true;
+        }
+      };
+      plugin.tracer.bind(call);
+      call.on('error', (err: grpcJs.ServiceError) => {
+        if (call[CALL_SPAN_ENDED]) {
+          return;
+        }
+        call[CALL_SPAN_ENDED] = true;
+
+        span.setStatus({
+          code: grpcStatusCodeToCanonicalCode(err.code),
+          message: err.message,
+        });
+        span.setAttributes({
+          [RpcAttribute.GRPC_ERROR_NAME]: err.name,
+          [RpcAttribute.GRPC_ERROR_MESSAGE]: err.message,
+        });
+
+        endSpan();
+      });
+
+      call.on('status', (status: Status) => {
+        if (call[CALL_SPAN_ENDED]) {
+          return;
+        }
+        call[CALL_SPAN_ENDED] = true;
+
+        span.setStatus(grpcStatusCodeToSpanStatus(status.code));
+
+        endSpan();
+      });
+    }
+    return call;
+  };
+}
+
+/**
+ * Returns the metadata argument from user provided arguments (`args`)
+ */
+function getMetadata(
+  this: GrpcJsPlugin,
+  original: GrpcClientFunc,
+  args: unknown[]
+): grpcJs.Metadata {
+  let metadata: grpcJs.Metadata;
+
+  // This finds an instance of Metadata among the arguments.
+  // A possible issue that could occur is if the 'options' parameter from
+  // the user contains an '_internal_repr' as well as a 'getMap' function,
+  // but this is an extremely rare case.
+  let metadataIndex = findIndex(args, (arg: grpcJs.Metadata) => {
+    return (
+      arg &&
+      typeof arg === 'object' &&
+      arg['internalRepr'] && // changed from _internal_repr in grpc --> @grpc/grpc-js
```

Comment:
For consistency with the field access below.

```suggestion
      arg.internalRepr && // changed from _internal_repr in grpc --> @grpc/grpc-js
```

Reply:
This is now marked as `protected` by grpc-js, so I need to access it "privately" via quotes

https://github.com/grpc/grpc-node/blob/95289edcaf36979cccf12797cc27335da8d01f03/packages/grpc-js/src/metadata.ts#L88